akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala

```diff
@@ -40,7 +42,15 @@ import scala.util.Try
 
   IO(Udp) ! Udp.Bind(self, new InetSocketAddress(InetAddress.getByAddress(Array.ofDim(4)), 0))
 
-  var inflightRequests: Map[Short, ActorRef] = Map.empty
+  var inflightRequests: Map[Short, (ActorRef, Message)] = Map.empty
+
+  val tcpDnsClient = context.actorOf(BackoffSupervisor.props(
+    Props(classOf[TcpDnsClient], ns, self),
+    childName = "tcpDnsClient",
+    minBackoff = 10.millis,
+    maxBackoff = 20.seconds,
+    randomFactor = 0.1
+  ), "tcpDnsClientSupervisor")
```

Comment:
I'm guessing this only happens for the SRV use case, not normal DNS, so should we perhaps start it lazily instead of always?

Reply:
In theory it can also happen for large normal DNS responses, but I agree that is a corner case and it'd make sense to start this lazily